### PR TITLE
[TEST] inject a random index to TestClusterService in BaseQueryTestCase#init

### PR DESCRIPTION
**Pull request by javanna**
Some of our next queries to refactor rely on some state taken from the cluster state. That is why we need to mock cluster service and inject an index to it, the index that we simulate the execution of the queries against. The best would be to have multiple indices actually, but that would make our setup a lot more complicated, especially given that IndexQueryParseService is still per index. We might be able to improve that in the future though, for now this is as good as it gets.

### CODE REVIEW

**Comment by alexksikes** [comment]
This looks good. I would also add a method to get the name of the index so we can use it in say Indices Query.

**Comment by javanna** [comment]
> I would also add a method to get the name of the index so we can use it in say Indices Query.

Makes sense, I had forgotten about that, I pushed an update.

File: core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java [hunk 1]
```diff
@@ -116,7 +122,10 @@ public static void init() throws IOException {
                         Version.V_1_0_0, Version.CURRENT))
                 .build();
 
-        index = new Index("test");
+        index = new Index(randomAsciiOfLengthBetween(1, 10));
+        final TestClusterService clusterService = new TestClusterService();
+        clusterService.setState(new ClusterState.Builder(clusterService.state()).metaData(new MetaData.Builder().put(
+                new IndexMetaData.Builder(index.name()).settings(Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)).numberOfShards(1).numberOfReplicas(0))));
```
Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java]: Just an idea, should the index maybe have the same version as the one we are randomly picking in the settings? Maybe not, I think I introduced it there to test both strict and non-strict mapping checks, but maybe we don't need that here.

Comment (javanna) [inline core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java]: sure we can do that, I didn't remember we were picking a version randomly to be honest, will look into that ;)

**Comment by cbuescher** [comment]
Also did a quick look, only one suggestion, otherwise LGTM

**Comment by javanna** [comment]
@cbuescher I pushed a new commit, hopefully sorted out index settings now. please have a look.

**Comment by cbuescher** [comment]
LGTM

**Comment by alexksikes** [comment]
LGTM

**Pull request merged**